### DISCUSSION TITLE
Show login page in new window

### DISCUSSION THREAD

**Discussion by Krm1t** [authelia/authelia#4083] [Ideas]
The idea comes from the fact that Authelia don't work when run inside an iframe for security reasons as i understand it.
So i thought what if the login page were to open a new window instead of showing in the iframe.
Wouldn't that allow sites that redirect to authelia to be shown in iframes while still be secure?

Ideally a new window would only open if required. So if the site isn't opened in an iframe it would just show the login page as it does now.

I believe Plex.tv (and other sites) does something similar when their login is being used on 3rd party sites.

**Comment by james-d-elliott** [comment]
The standard redirection flow that doesn't require specific support from an application is limited in as much as the user must be redirected for it to work. We can't create a new window with this unless I'm missing something. For a popup or new window I believe we need to have javascript injected into the target application.

The protections Authelia implements can technically be turned off to allow iFrames. However disabling this protection makes it possible for an attacker (and this is true with ANY system that doesn't block iFrmes and XSS) to make a website look like a website a user is familiar with to steal their credentials. 

The way Plex does this is via OpenID Connect popups from memory. I think (but am unsure) that the relying party (client) must explicitly support this.
